addons/rating/static/src/js/rating_portal.js

```diff
@@ -0,0 +1,25 @@
+/** @odoo-module alias=portal.rating_portal **/
+
+import publicWidget from 'web.public.widget';
+
+publicWidget.registry.ratingPortal = publicWidget.Widget.extend({
+    selector: '.o_rating_page_submit',
+    events: {
+        'click .o_rating_label': '_onRatingLabelClick',
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onRatingLabelClick: function (ev) {
+        const radioButtons = document.querySelectorAll('.o_rating_label');
+        radioButtons.forEach(button => {
+            button.classList.remove('active');
+        });
+        ev.currentTarget.classList.add("active");
+    },
+});
```

Comment:
All the rating related code can be removed as already handled by the SM team (see here https://github.com/odoo/odoo/pull/95623)

@aju-odoo 👆🏼